conan/cli/formatters/audit/vulnerabilities.py

```diff
@@ -12,6 +13,11 @@
 }
 
 
+def _raise_if_severity_exceeded(result: tuple[dict, bool]):
+    if result[0].get("is_error_severity", False):
+        raise ConanException("The conan audit command exceeded the threshold severity level.")
```

Comment:
There is a mechanism in the formatters. If the result of contains an ``error`` field, Conan will automatically raise after the formatters with that error.

Reply:
The audit uses a separate formatter: https://github.com/conan-io/conan/blob/develop2/conan/cli/formatters/audit/vulnerabilities.py#L15

Still, after your comment I updated the code to follow what conan provider is doing: Set errors.details and raise an error. Check the commit [573ae7a](https://github.com/conan-io/conan/pull/18160/commits/573ae7ae29cb09c3c18ec0eb9f386bff4e6346cf)